elfo-network/src/frame/lz4.rs

```diff
@@ -0,0 +1,138 @@
+//! This module implements custom LZ4 framing.
+//!
+//! Single LZ4 frame contains one or more envelopes compressed as a single
+//! LZ4 block along with some meta information like frame size. We do NOT
+//! use standard LZ4 framing.
+//!
+//! Structure of a single frame:
+//!           name           bits
+//! +---------------------------+----+
+//! | size of whole frame       | 32 |
+//! +---------------------------+----+
+//! | size of decompressed data | 32 |
+//! +---------------------------+----+
+//! | LZ4 block                 |rest|
+//! +---------------------------+----+
+//!
+//! All fields are encoded using LE ordering.
+
+// TODO: checksums.
+
+use byteorder::{LittleEndian, ReadBytesExt, WriteBytesExt};
+use std::io::Cursor;
+
+use eyre::{eyre, Result};
+
+use crate::codec::decode::DecodeState;
+
+pub(crate) struct LZ4Buffer {
+    buffer: Vec<u8>,
```

Comment:
This field changes semantics depending on the usage: it stores compressed data in one case and uncompressed in another, it should be commented.